src/util/storage.js

```diff
@@ -1,10 +1,40 @@
+import { isWindowDefined } from './clevertap'
 import {
   GCOOKIE_NAME,
   META_COOKIE,
   KCOOKIE_NAME,
   LCOOKIE_NAME
 } from './constants'
+import { getHostName } from './url'
+
 export class StorageManager {
+  /**
+   * @type {('sync' | 'async')} saveMode
+   */
+  static saveMode = 'sync';
```

Comment:
Going with this approach.
The idea is that if we are going to run our SDK in different environment, there will be different ways to store data in LS. In shopify we return a promise while storing data in LS. Hence we set `saveMode` to `async` to indicate that we will call all the async functions for saving and reading.